parity/configuration.rs

```diff
@@ -666,23 +666,32 @@ impl Configuration {
 		}
 
 		let usd_per_tx = to_price(&self.args.arg_usd_per_tx)?;
-		if "auto" == self.args.arg_usd_per_eth.as_str() {
-			return Ok(GasPricerConfig::Calibrated {
-				usd_per_tx: usd_per_tx,
-				recalibration_period: to_duration(self.args.arg_price_update_period.as_str())?,
-			});
-		}
 
-		let usd_per_eth = to_price(&self.args.arg_usd_per_eth)?;
-		let wei_per_gas = wei_per_gas(usd_per_tx, usd_per_eth);
+		let usd_per_eth = to_price(&self.args.arg_usd_per_eth);
+		match usd_per_eth {
+			Ok(usd_per_eth_parsed) => {
+				let wei_per_gas = wei_per_gas(usd_per_tx, usd_per_eth_parsed);
 
-		info!(
-			"Using a fixed conversion rate of Ξ1 = {} ({} wei/gas)",
-			Colour::White.bold().paint(format!("US${:.2}", usd_per_eth)),
-			Colour::Yellow.bold().paint(format!("{}", wei_per_gas))
-		);
+				info!(
+					"Using a fixed conversion rate of Ξ1 = {} ({} wei/gas)",
+					Colour::White.bold().paint(format!("US${:.2}", usd_per_eth_parsed)),
+					Colour::Yellow.bold().paint(format!("{}", wei_per_gas))
+				);
 
-		Ok(GasPricerConfig::Fixed(wei_per_gas))
+				return Ok(GasPricerConfig::Fixed(wei_per_gas))
+			},
+			Err(_) => {
+				let mut arg_usd_per_eth_endpoint = self.args.arg_usd_per_eth.as_str();
+				if "auto" ==  arg_usd_per_eth_endpoint {
+					arg_usd_per_eth_endpoint = "https://api.etherscan.io/api?module=stats&action=ethprice"
```

Comment:
I believe this default value should go into some kind of a defaults module... will try to check if there is one.

Reply:
yeah, it's a bit nasty to have hardcoded strings like this. Even a `const` would be an improvement.

Reply:
I'm not super intimate with this piece,  however, I think it would be more clear to explicit check `if "auto" == self.args.arg_usd_per_eth.as_str()` before matching/parsing the string as a floating-point number.

In addition, I don't think it makes sense to add an `API endpoint` to  `GasPricerConfig::Calibrated` because it is already stored in `PriceInfoClient`? Why is it needed?

Reply:
Used if-let and defined the endpoint as a constant.